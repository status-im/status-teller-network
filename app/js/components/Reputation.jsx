import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Card } from 'reactstrap';
import { withNamespaces } from 'react-i18next';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faThumbsDown, faThumbsUp } from "@fortawesome/free-solid-svg-icons";

class Reputation extends Component {
  render() {
    return (
      <Row className="mt-4">
        <Col xs="12">
          <span className="font-weight-bold h5">{this.props.t('reputation.title')}</span>
        </Col>
        <Col xs="6" className="pr-0">
          <Card body className="text-center rounded-0">
            <div>
              <span className="mr-3">{this.props.reputation.upCount}</span>
              <FontAwesomeIcon className="text-warning" size="lg" icon={faThumbsUp}/>
            </div>
          </Card>
        </Col>
        <Col xs="6" className="pl-0">
          <Card body className="text-center rounded-0">
            <div>
              <span className="mr-3">{this.props.reputation.downCount}</span>
              <FontAwesomeIcon className="text-warning" size="lg" icon={faThumbsDown}/>
            </div>
          </Card>
        </Col>
      </Row>
    );
  }
}

Reputation.propTypes = {
  t: PropTypes.func,
  reputation: PropTypes.object
};

export default withNamespaces()(Reputation);
