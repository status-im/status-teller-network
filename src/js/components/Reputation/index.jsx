/* eslint-disable no-confusing-arrow */
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {withNamespaces} from 'react-i18next';
import classnames from "classnames";

import RatingIcon from "../../ui/RatingIcon";

import './index.scss';

class Reputation extends Component {
  rateTrade(rating) {
    if (!this.clickable) {
      return;
    }
    this.props.rateTransaction(this.props.trade.escrowId, rating, this.props.isBuyer);
  }

  /*eslint-disable-next-line complexity */
  render() {
    const {size, rateTransaction, trade, isBuyer, reputation = {}} = this.props;
    const rating = trade ? parseInt(isBuyer ? trade.sellerRating : trade.buyerRating, 10) : 0;
    const tradeWasRated = trade && rating !== 0;
    this.clickable = !tradeWasRated && rateTransaction && trade;

    if (reputation.hasOwnProperty('averageCount')) {
      let classes = 'average-rating py-1 px-3 rounded ';
      if (!Number.isInteger(reputation.averageCount)) {
        classes += 'bg-dark text-dark';
      } else  if (reputation.averageCount >= 8) {
        classes += 'bg-success text-success';
      } else  if (reputation.averageCount < 8 && reputation.averageCount > 4) {
        classes += 'bg-warning text-warning';
      } else  if (reputation.averageCount < 4) {
        classes += 'bg-danger text-danger';
      }

      return <span className={classes}>
        {Number.isInteger(reputation.averageCount) ? reputation.averageCount : 'NA'}
      </span>;
    }

    return <span className={classnames("reputation-container", {small: size === 's', large: size === 'l'})}>
      <span className={classnames("left-rating", {'rounded-circle pt-2 bg-success rating-with-action': this.clickable || tradeWasRated, 'clickable': this.clickable})}>
      {(tradeWasRated && reputation.upCount) || (!trade && reputation.upCount)}
        &nbsp;<RatingIcon isPositiveRating={true} isRated={tradeWasRated && rating === 5} onClick={this.clickable ? () => this.rateTrade('5') : null} size={this.clickable || tradeWasRated ? 'lg' : 'sm'}/>
      </span>
      <span className={classnames("right-rating", {' bg-danger rounded-circle pt-2 rating-with-action': this.clickable || tradeWasRated, 'clickable': this.clickable})}>
      {(tradeWasRated && reputation.downCount) || (!trade && reputation.downCount)}
        &nbsp;<RatingIcon isPositiveRating={false} isRated={tradeWasRated && rating === 1} onClick={this.clickable ? () => this.rateTrade('1') : null} size={this.clickable || tradeWasRated ? 'lg' : 'sm'}/>
      </span>
    </span>;
  }
}

Reputation.propTypes = {
  t: PropTypes.func,
  size: PropTypes.string,
  isBuyer: PropTypes.bool,
  trade: PropTypes.object,
  rateTransaction: PropTypes.func,
  reputation: PropTypes.object
};

export default withNamespaces()(Reputation);
