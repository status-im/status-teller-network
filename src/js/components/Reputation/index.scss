@import "../../../css/variable-overrides";

.reputation-container {
  .left-rating, .right-rating {
    font-size: 15px;
    margin: 0 1px;
    display: inline-block;
    text-align: center;
  }

  .left-rating {
    margin-right: 20px;
  }

  .rating-with-action {
    width: 40px;
    height: 40px;
  }

  &.small {
    .left-rating, .right-rating {
      font-size: 13px;
      width: 55px;
      height: 24px;
      line-height: 22px;
    }
  }
}

.average-rating, .rating-with-action {
  $rating-opacity-bg: 38%;

  &.bg-dark {
    background-color: lighten($dark, $rating-opacity-bg) !important;
  }

  &.bg-success {
    background-color: lighten($success, $rating-opacity-bg) !important;
  }

  &.bg-warning {
    background-color: lighten($warning, $rating-opacity-bg) !important;
  }

  &.bg-danger {
    background-color: lighten($danger, $rating-opacity-bg) !important;
  }
}
