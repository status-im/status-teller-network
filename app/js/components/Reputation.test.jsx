import React from "react";
import { shallow } from "enzyme";

import Reputation from "./Reputation";

describe('Trades', () => {
  it('should render correctly', () => {
    const component = shallow(<Reputation reputation={{upCount: 1, downCount: 2}} />);
  
    expect(component).toMatchSnapshot();
  });
});
