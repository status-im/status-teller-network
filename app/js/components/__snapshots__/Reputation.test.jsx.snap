// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trades should render correctly 1`] = `
<Row
  className="mt-4"
  tag="div"
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="12"
  >
    <span
      className="font-weight-bold h5"
    />
  </Col>
  <Col
    className="pr-0"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="6"
  >
    <Card
      body={true}
      className="text-center rounded-0"
      tag="div"
    >
      <div>
        <span
          className="mr-3"
        >
          1
        </span>
        <FontAwesomeIcon
          border={false}
          className="text-warning"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f164",
                "M104 224H24c-13.255 0-24 10.745-24 24v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V248c0-13.255-10.745-24-24-24zM64 472c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zM384 81.452c0 42.416-25.97 66.208-33.277 94.548h101.723c33.397 0 59.397 27.746 59.553 58.098.084 17.938-7.546 37.249-19.439 49.197l-.11.11c9.836 23.337 8.237 56.037-9.308 79.469 8.681 25.895-.069 57.704-16.382 74.757 4.298 17.598 2.244 32.575-6.148 44.632C440.202 511.587 389.616 512 346.839 512l-2.845-.001c-48.287-.017-87.806-17.598-119.56-31.725-15.957-7.099-36.821-15.887-52.651-16.178-6.54-.12-11.783-5.457-11.783-11.998v-213.77c0-3.2 1.282-6.271 3.558-8.521 39.614-39.144 56.648-80.587 89.117-113.111 14.804-14.832 20.188-37.236 25.393-58.902C282.515 39.293 291.817 0 312 0c24 0 72 8 72 81.452z",
              ],
              "iconName": "thumbs-up",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </div>
    </Card>
  </Col>
  <Col
    className="pl-0"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="6"
  >
    <Card
      body={true}
      className="text-center rounded-0"
      tag="div"
    >
      <div>
        <span
          className="mr-3"
        >
          2
        </span>
        <FontAwesomeIcon
          border={false}
          className="text-warning"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f165",
                "M0 56v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56zm40 200c0-13.255 10.745-24 24-24s24 10.745 24 24-10.745 24-24 24-24-10.745-24-24zm272 256c-20.183 0-29.485-39.293-33.931-57.795-5.206-21.666-10.589-44.07-25.393-58.902-32.469-32.524-49.503-73.967-89.117-113.111a11.98 11.98 0 0 1-3.558-8.521V59.901c0-6.541 5.243-11.878 11.783-11.998 15.831-.29 36.694-9.079 52.651-16.178C256.189 17.598 295.709.017 343.995 0h2.844c42.777 0 93.363.413 113.774 29.737 8.392 12.057 10.446 27.034 6.148 44.632 16.312 17.053 25.063 48.863 16.382 74.757 17.544 23.432 19.143 56.132 9.308 79.469l.11.11c11.893 11.949 19.523 31.259 19.439 49.197-.156 30.352-26.157 58.098-59.553 58.098H350.723C358.03 364.34 384 388.132 384 430.548 384 504 336 512 312 512z",
              ],
              "iconName": "thumbs-down",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </div>
    </Card>
  </Col>
</Row>
`;
